services/ai-engine/main.py

```
@@ -0,0 +1,116 @@
+    if file.content_type != "application/pdf":
+        raise HTTPException(status_code=400, detail="Unsupported file type. Only PDF is allowed.")
+    
+    temp_path = await _read_bounded_to_disk(file, MAX_FILE_SIZE)
+    
+    try:
+        if os.path.getsize(temp_path) == 0:
+            raise HTTPException(status_code=400, detail="Empty file uploaded.")
+            
+        try:
+            doc = fitz.open(temp_path)
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)